### PR TITLE
[ci] Add LUCI repo tool tests

### PR DESCRIPTION
**Pull request by stuartmorgan**
- Adds a LUCI version of the Linux repo tools test, as a first test of a Linux repository test being migrated to LUCI.
- Fixes the Windows repo tools test to actually run a test, which it wasn't due to a mistaken duplicate yaml file, only one of which was correct (the one that wasn't being used).

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -44,7 +44,18 @@ platform_properties:
       xcode: 14e222b
 
 targets:
-  ### iOS+macOS tasks ***
+  ### Linux tasks ###
+  - name: Linux repo_tools_tests
```
Comment (keyonghan) [inline .ci.yaml:48]: Does this test need the docker image support?
/cc @yusuf-goog who is helping with general docker support for vm based builds.

Comment (stuartmorgan) [reply .ci.yaml:48]: I would not expect it to.

Per [this comment](https://github.com/flutter/flutter/issues/120732#issuecomment-1430589359), I'm not convinced we need docker support at all for this repo.

Comment (stuartmorgan) [reply .ci.yaml:48]: (This test is literally just running `dart test` in a directory, so if we need Docker for this something is wrong.)

Comment (keyonghan) [reply .ci.yaml:48]: Nice. Then the Linux migration should be unblocked! /cc @yusuf-goog who is also helping with remaining packages Linux tasks migration to LUCI.

One thing to check is whether this test needs any dependencies. Now the `linux` platform only enforces the os dimension `os: Linux`. An LED run may give you some signal. This is the doc how to test a new builder: go/flutter-luci-recipes#testing-a-local-change-for-a-new-builderrecipe. For this case specifically, you can use `Windows repo_tools_tests`, and change the `os` to Linux.

Comment (stuartmorgan) [reply .ci.yaml:48]: https://ci.chromium.org/raw/build/logs.chromium.org/flutter/led/stuartmorgan_google.com/b340388546ffb86941a2a5b4307cf8c79adbc9436acda46dc29f0a7b72a1068c/+/build.proto?server=chromium-swarm.appspot.com

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**